test/Table_Tests/src/Common_Table_Operations/Missing_Values_Spec.enso

```diff
@@ -218,12 +218,19 @@ spec setup =
             table.at "Z" . fill_nothing True . should_fail_with No_Common_Type
 
         if setup.is_database.not then
-            Test.specify "may allow mixed types if explicitly retyped" pending="ToDo: cast #6112" <|
+            Test.specify "may allow mixed types if explicitly retyped" <|
                 table = table_builder [["X", [1, Nothing, 2, Nothing]]]
-                ints = table.at "X" . cast Value_Type.Mixed
-                ints_filled = ints.fill_nothing False
-                ints_filled.to_vector . should_equal [1, False, 2, False]
-                ints_filled.value_type . should_equal Value_Type.Mixed
+                mix = table.at "X" . cast Value_Type.Mixed
+                mix_filled = mix.fill_nothing False
+                mix_filled.to_vector . should_equal [1, False, 2, False]
+                mix_filled.value_type . should_equal Value_Type.Mixed
+
+            Test.specify "will keep the Mixed type if was explicitly retyped" <|
+                table = table_builder [["X", [1, Nothing, 2, Nothing]]]
+                mix = table.at "X" . cast Value_Type.Mixed
```

Comment:
Does this behave differently if there is no explicit `cast`?

Reply:
Currently it doesn't but it's supposed to.

Currently (and without cast), it fails with: `No_Common_Type.Error [(Integer 64 bits), Boolean]` - because the types Integer and Boolean are not compatible.

But once we cast to Mixed, any mix of types should be accepted. And currently that does not work, but it should - I'm meaning to fix it in a moment.